src/main/java/io/cdap/plugin/servicenow/sink/service/ServiceNowSinkAPIRequestImpl.java

```diff
@@ -0,0 +1,197 @@
+  private static final String INSERT_TABLE_API_URL_TEMPLATE = "/api/now/table/%s";
+  private static final String UPDATE_TABLE_API_URL_TEMPLATE = "/api/now/table/%s/%s";
+  private static final String HTTP_POST = "POST";
+  private static final String HTTP_PUT = "PUT";
+  private static final String SYS_ID = "sys_id";
+  private static final String SERVICED_REQUESTS = "serviced_requests";
+  private static final String UNSERVICED_REQUESTS = "unserviced_requests";
+  private static Integer counter = 1;
+  private static Integer batchRequestIdCounter = 1;
+  private static Integer retryCounter = 0;
+  private final ServiceNowSinkConfig config;
+  private final ServiceNowTableAPIClientImpl restApi;
+
+  public ServiceNowSinkAPIRequestImpl(ServiceNowSinkConfig conf) {
+    this.config = conf;
+    restApi = new ServiceNowTableAPIClientImpl(config);
+  }
+
+  public RestRequest getRestRequest(JsonObject jsonObject) {
+    JsonElement jsonElement = jsonObject.get(SYS_ID);
+    String sysId = null;
+
+    if (jsonElement == null) {
+      if (config.getOperation().equals("update")) {
+        throw new RuntimeException("No sys_id found in the record to be updated");
+      }
+    } else {
+      if (config.getOperation().equals("insert")) {
+        jsonObject.remove(SYS_ID);
+      }
+      sysId = jsonElement.getAsString();
+    }
+
+    String data = jsonObject.toString();
+    String encodedData = Base64.getEncoder().encodeToString(data.getBytes(StandardCharsets.UTF_8));
+
+    List<Header> headers = new ArrayList<>();
+    Header contentTypeHeader = new BasicHeader("Content-Type", "application/json");
+    Header acceptHeader = new BasicHeader("Accept", "application/json");
+    headers.add(contentTypeHeader);
+    headers.add(acceptHeader);
+
+    RestRequest restRequest = new RestRequest();
+    restRequest.setUrl(String.format(INSERT_TABLE_API_URL_TEMPLATE, config.getTableName()));
+    if (config.getOperation().equals("update")) {
+      restRequest.setUrl(String.format(UPDATE_TABLE_API_URL_TEMPLATE, config.getTableName(), sysId));
+    }
+    restRequest.setId(counter.toString());
+    counter++;
+    restRequest.setHeaders(headers);
+    restRequest.setMethod(HTTP_POST);
+    if (config.getOperation().equals("update")) {
+      restRequest.setMethod(HTTP_PUT);
+    }
+    restRequest.setBody(encodedData);
+    return restRequest;
+  }
+
+  public Boolean createPostRequest(List<RestRequest> records) {
+    ServiceNowBatchRequest payloadRequest = getPayloadRequest(records);
+    ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+      config.getRestApiEndpoint());
+    RestAPIResponse apiResponse;
+
+    try {
+      String accessToken = restApi.getAccessToken();
+      requestBuilder.setAuthHeader(accessToken);
+      requestBuilder.setAcceptHeader("application/json");
+      requestBuilder.setContentTypeHeader("application/json");
+      Gson gson = new Gson();
```

Comment:
This can be declared as class level private final

Reply:
Declared as a class level private final variable.